src/main/java/ua/softserveinc/tc/validator/KidProfileImageValidatorIml.java

```diff
@@ -31,62 +36,117 @@
     private ApplicationConfigurator applicationConfigurator;
 
     @Override
-    public boolean isNotNullOrEmpty(MultipartFile file) {
-        return file != null && !file.isEmpty();
-    }
-
-    @Override
-    public boolean isHaveAcceptableSize(MultipartFile file) {
-        return file.getSize() <= applicationConfigurator.getMaxUploadImgSizeMb()
-                * ImageConstants.ONE_MEGA_BYTE_IN_BYTES;
+    public boolean isCorrupted(MultipartFile file) {
+        BufferedImage image = null;
+        ImageReader imgReader = getImageReader(file);
+        if(imgReader != null) {
+            try (ImageInputStream inStream =
+                         ImageIO.createImageInputStream(file.getInputStream())) {
+                imgReader.setInput(inStream);
+                image = imgReader.read(0);
+            } catch (IOException e) {
+                log.error("Failed to read image from file : " + file, e);
+                image = null;
+            }
+        }
+        return image == null;
     }
 
     @Override
-    public boolean isHaveAcceptableFormat(MultipartFile file) {
+    public boolean isAcceptableSize(MultipartFile file) {
         boolean result = false;
-        try (ImageInputStream iis =
-                ImageIO.createImageInputStream(file.getInputStream())) {
-            Iterator<ImageReader> readers = ImageIO.getImageReaders(iis);
-
-            out:
-            while (readers.hasNext()) {
-                ImageReader reader = readers.next();
-                for (String imgFo : applicationConfigurator.getImageAcceptableFormats())
-                    if (imgFo.toLowerCase().equals(reader.getFormatName().toLowerCase())) {
-                        result = true;
-                        break out;
-                    }
-            }
-        } catch(IOException e) {
-            log.error("Failed to save image", e);
+        if(file != null) {
+            result = file.getSize() <= applicationConfigurator.getMaxUploadImgSizeMb()
+                    * ImageConstants.ONE_MEGA_BYTE_IN_BYTES;
         }
         return result;
     }
 
     @Override
+    public boolean isAcceptableFormat(MultipartFile file) {
+        return isAcceptableFormat(getImageReader(file));
+    }
+
+    @Override
     public void validate(Object o, Errors errors) {
         if (!(o instanceof MultipartFile)) {
             log.error("Validator error! Entered wrong object for validation");
             errors.rejectValue(ValidationConstants.IMAGE,
                     ValidationConstants.IMAGE_VALIDATION_NOT_CORRECT_USAGE);
         } else {
-            if (!isNotNullOrEmpty((MultipartFile)o))
+            if (((MultipartFile)o).isEmpty()) {
```

Comment:
Think it's better to perform type conversion one time at beginning of the block:
 MultipartFile file = (MultipartFile) o;